From 1b33676f9724837f5d29dee79306278a86480d8a Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Thu, 23 Feb 2012 07:57:29 +0100
Subject: [PATCH linphone 1/2] Allow speex support to be explicitly disabled

Some installations may prefer not to add support for the speex codec.
The new --disable-speex configure option allows support to be explicitly
disabled while keeping it enabled by default.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 configure.ac |   18 +++++++++++-------
 1 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 22bc235..67060be 100644
--- a/configure.ac
+++ b/configure.ac
@@ -330,13 +330,17 @@ fi
 
 AC_WORDS_BIGENDIAN
 
-dnl normaly this should only by done by mediastreamer2/configure.ac
-dnl but to workaround bugs when cross-compiling for arm-linux,
-dnl we need to have SPEEX_LIBS defined
-dnl Furthermore it is good to repeat here all mediastreamer2 toggles
-dnl since top-level configure --help will not print them.
-
-PKG_CHECK_MODULES(SPEEX, speex >= 1.1.6, build_speex=yes)
+AC_ARG_ENABLE([speex],
+	      AS_HELP_STRING([--disable-speex], [Disable speex support]),
+	      [], [enable_speex=yes])
+if test "x$enable_speex" = "xyes"; then
+	dnl normaly this should only by done by mediastreamer2/configure.ac
+	dnl but to workaround bugs when cross-compiling for arm-linux,
+	dnl we need to have SPEEX_LIBS defined
+	dnl Furthermore it is good to repeat here all mediastreamer2 toggles
+	dnl since top-level configure --help will not print them.
+	PKG_CHECK_MODULES(SPEEX, speex >= 1.1.6, build_speex=yes)
+fi
 
 dnl conditionnal build of video support
 AC_ARG_ENABLE(video,
-- 
1.7.9.1

